src/CoreShop/Bundle/OrderBundle/Pimcore/Repository/OrderRepository.php

```diff
@@ -40,7 +40,7 @@ public function findCartByCustomer(CustomerInterface $customer): array
         return $carts;
     }
 
-    public function findByCartId(int $id): ?OrderInterface
+    public function findByCartId(?int $id): ?OrderInterface
```

Comment:
Maybe you should return early in this case, as `o_id` afaik cannot be `null` and we don't need to query the database in this case.